### PR TITLE
RPi add RPi and RPi2 devices

### PR DESCRIPTION
**Pull request by lrusak**
This allows us to remove a lot of redundant code.

requires #1356 

RPi projects will now have to be built with:
```
PROJECT=RPi DEVICE=RPi2 ARCH=arm make
PROJECT=RPi DEVICE=RPi ARCH=arm make
```

### CODE REVIEW

File: projects/RPi/options [hunk 1]
```diff
@@ -17,7 +17,12 @@
         # arm1136j-s arm1136jf-s mpcore mpcorenovfp arm1156t2-s
         # arm1176jz-s arm1176jzf-s cortex-a8 cortex-a9 cortex-r4
         # cortex-r4f cortex-m3 cortex-m1 xscale iwmmxt iwmmxt2 ep9312.
-        TARGET_CPU="arm1176jzf-s"
+
+        if [ "$DEVICE" = "RPi" ]; then
+          TARGET_CPU="arm1176jzf-s"
+        elif [ "$DEVICE" = "RPi2" ]; then
+          TARGET_CPU="cortex-a7"
+        fi
```
Comment (kszaq) [inline projects/RPi/options]: Why not define `TARGET_CPU` in `projects/RPi/device/*/options`?

**Comment by lrusak** [comment]
would be nice to get this in before it diverges too much

File: projects/RPi/options (hunk 1 above)
Comment (lrusak) [inline projects/RPi/options]: I actually considered this, however then you have more dangling files to manage.

**Comment by lrusak** [comment]
rebased.

**Comment by MilhouseVH** [comment]
Can you rebase again (after #1400) - thanks!

**Comment by MilhouseVH** [comment]
Before merging we'll need to work out a solution for the compatibility check, also if there are any custom update channel changes needed. I'll look into both and ping you any changes.

**Comment by MilhouseVH** [comment]
I think this patch will allow old project-only systems to upgrade: http://sprunge.us/TAGP

**Comment by MilhouseVH** [comment]
This patch will set OPENELEC_ARCH and LIBREELEC_ARCH to appropriate values when building for a device: http://sprunge.us/HSaB

Unless you can think of a better/more appropriate solution?

**Comment by MilhouseVH** [comment]
For now, unless you can think of any issues, I'd recommend:

1) Add http://sprunge.us/iDiW to this PR
2) Add http://sprunge.us/OjFM to libreelec-8.0 before 8.0.1 (I can do that if nobody objects - whoever is responsible for Slice will need it also)
3) Add http://sprunge.us/HSaB to this PR, but backend changes may also be required (auto-update, custom update).

**Comment by lrusak** [comment]
rebased

**Comment by lrusak** [comment]
rebased and good to go after #1356

**Comment by lrusak** [comment]
rebased

**Pull request merged**